### PR TITLE
NDRCN-232

### PR DESCRIPTION
**Pull request by simms11**

### CODE REVIEW

**Comment by simms11** [comment]
Navigation has not been added as this page is  specifically related to the **representative of the importer journey** which has yet to be completed

**Review by lmugnai — CHANGES_REQUESTED** [review]
Need some specs for this page.

File: app/controllers/ProofOfAuthorityController.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                                            view: ProofOfAuthorityView
+                                          )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      Ok(view())
+  }
+
+}
```
Comment (lmugnai) [inline app/controllers/ProofOfAuthorityController.scala:43]:
```suggestion
}

```

File: app/views/ProofOfAuthorityView.scala.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <p class="mt-0">@messages("proofOfAuthority.paragraph2")</p>
+
+
+    <div class="section">
+        <a id = "govuk-link" class="button" role = "button" href="@routes.BankDetailsController.onPageLoad(NormalMode)">
+        @messages("site.continue")
+        </a>
+    </div>
+
+}
```
Comment (lmugnai) [inline app/views/ProofOfAuthorityView.scala.html:45]:
```suggestion
}

```

File: conf/app.routes [hunk 3]
```diff
@@ -233,3 +233,5 @@ GET        /indirectRepresentative                        controllers.IndirectRe
 POST       /indirectRepresentative                        controllers.IndirectRepresentativeController.onSubmit(mode: Mode = NormalMode)
 GET        /changeIndirectRepresentative                  controllers.IndirectRepresentativeController.onPageLoad(mode: Mode = CheckMode)
 POST       /changeIndirectRepresentative                  controllers.IndirectRepresentativeController.onSubmit(mode: Mode = CheckMode)
+
+GET         /proofOfAuthority                       controllers.ProofOfAuthorityController.onPageLoad()
```
Comment (lmugnai) [inline conf/app.routes:237]:
```suggestion
GET         /proofOfAuthority                       controllers.ProofOfAuthorityController.onPageLoad()

```

File: conf/messages.en [hunk 4]
```diff
@@ -408,3 +408,8 @@ indirectRepresentative.title = Are you an indirect representative of the importe
 indirectRepresentative.heading = Are you an indirect representative of the importer?
 indirectRepresentative.checkYourAnswersLabel = Are you an indirect representative of the importer?
 indirectRepresentative.error.required = Select yes if are you an indirect representative of the importer
+
+proofOfAuthority.title = Upload your proof of authority
+proofOfAuthority.heading = Upload your proof of authority
+proofOfAuthority.paragraph1 = You must show that the importer has given you authority to receive payment for this specific application
+proofOfAuthority.paragraph2 = Upload a file
```
Comment (lmugnai) [inline conf/messages.en:415]:
```suggestion
proofOfAuthority.paragraph2 = Upload a file

```

**Review by lmugnai — CHANGES_REQUESTED** [review]
Hi @simms11 Could you ensure there is a new line character at the end of each file? I've put them in as suggestions so just need to accept the suggestion. Thanks

File: app/navigation/Navigator.scala [hunk 5]
```diff
@@ -73,7 +74,7 @@ class Navigator @Inject()() {
   }
 
   private def indirectRepresentativeRoute(answers: UserAnswers): Call = answers.get(IndirectRepresentativePage) match {
-    case Some(true)  => routes.BankDetailsController.onPageLoad(NormalMode)
+    case Some(true)  => routes.ProofOfAuthorityController.onPageLoad()
```
Comment (lmugnai) [inline app/navigation/Navigator.scala:77]: I think this is wrong. If the user selects 'Yes' they should go to the BankDetails Page. If they select No they should go to the proofofauthority page. That's why I left the case(false) as not implemented (???).

File: test/pages/ProofOfAuthorityPageSpec.scala [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package pages
+
+import models.EvidenceSupportingDocs
+import pages.behaviours.PageBehaviours
+
+class ProofOfAuthorityPageSpec extends PageBehaviours {
+
+  "EvidenceSupportingDocsPage" must {
```
Comment (lmugnai) [inline test/pages/ProofOfAuthorityPageSpec.scala:24]: If this does nothing, can we remove it?

**Review by lmugnai — CHANGES_REQUESTED** [review]

File: test/navigation/NavigatorSpec.scala [hunk 7]
```diff
@@ -65,6 +68,26 @@ class NavigatorSpec extends SpecBase {
...
+        val answers =
+          emptyUserAnswers
+            .set(ClaimantTypePage, ClaimantType.Representative).success.value
+            .set(IndirectRepresentativePage, false).success.value
+
+        navigator.nextPage(IndirectRepresentativePage, NormalMode, answers)
+          .mustBe(routes.ProofOfAuthorityController.onPageLoad)
+      }
+
+      "go to CheckYourAnswers page after the claimant page once the representative has uploaded their proof of authority" in {
```
Comment (lmugnai) [inline test/navigation/NavigatorSpec.scala:82]: Can we have a chat about this test? It doesn't look right to me but might be missing something?

**Review by lmugnai — APPROVED** [review]

**Pull request merged**